xarray/tests/test_backends_datatree.py

```diff
@@ -337,6 +337,32 @@ def test_open_groups_to_dict(self, tmpdir) -> None:
         for ds in aligned_dict_of_datasets.values():
             ds.close()
 
+    def test_open_datatree_specific_group(self, tmpdir, simple_datatree) -> None:
+        """Test opening a specific group within a NetCDF file using `open_datatree`."""
+        filepath = tmpdir / "test.nc"
+        group = "/set1"
+        original_dt = simple_datatree
+        original_dt.to_netcdf(filepath)
+
+        with open_datatree(filepath, group=group, engine=self.engine) as subgroup_tree:
+            assert subgroup_tree is not None
+            assert subgroup_tree.root.parent is None
+            assert list(subgroup_tree.children) == list(original_dt[group].children)
+            assert len(subgroup_tree.dataset.dims) == len(
+                original_dt[group].dataset.dims
+            )
+            assert list(subgroup_tree.dataset.data_vars) == list(
+                original_dt[group].dataset.data_vars
+            )
+            assert (
+                subgroup_tree.dataset["a"].values
+                == original_dt[group].dataset["a"].values
+            )
+            assert (
+                subgroup_tree.dataset["b"].values
+                == original_dt[group].dataset["b"].values
+            )
```

Comment:
same here: replace with `assert_equal`?

Reply:
Could we not just use one top-level `assert_equal` on the tree itself? i.e.

```python
expected_subtree = original_dt[group].orphan()
assert_equal(subgroup_tree, expected_subtree)
```

Reply:
yes, we could, except `orphan` works in-place and modifies the original tree (it's one of the few remaining places that does so – should we change that before the release?).

Reply:
I'm not completely sure if it should work in-place or not. It doesn't violate @shoyer 's rule about working in-place and returning a new result, because it returns `None`.

Let's use `.orphan` as I'm suggesting here to simplify this test, then we can modify its behaviour (and therefore also this test) if we want to in a follow-up PR.

Reply:
then this:
```python
expected_subtree = original_dt[group].orphan()
```
will have to change to
```python
expected_subtree = original_dt[group]
expected_subtree.orphan()
```

Reply:
So instead my suggestion should have been

```python
expected_subtree = original_dt[group].copy()
expected_subtree.orphan()

assert subgroup_tree.root.parent is None
assert_equal(subgroup_tree, expected_subtree)
```

Reply:
orphan returns None. Therefore, both datatree cannot be compared. Am I wrong?

Reply:
(simultaneous comment - we agree)

Reply:
> orphan returns `None`. Therefore, both datatree cannot be compared. Am I wrong?

Orphan does return `None`, but it modifies the tree it's called upon. (That's what @keewis means by "in-place".) Our proposed code above should still work, because it compares the modified tree.